src/models/action.rs

```diff
@@ -1,26 +1,90 @@
 use chrono::NaiveDateTime;
+use diesel::prelude::*;
+use diesel::{
+    deserialize::{self, FromSql},
+    pg::Pg,
+    serialize::{self, Output, ToSql},
+    sql_types::Integer,
+};
+use std::io::Write;
 
 use crate::models::{ApiToken, User, Version};
 use crate::schema::*;
 
-#[derive(Debug, Clone, Copy)]
-#[repr(u32)]
+#[derive(Debug, Clone, Copy, PartialEq, FromSqlRow, AsExpression)]
+#[repr(i32)]
+#[sql_type = "Integer"]
 pub enum VersionAction {
     Publish = 0,
     Yank = 1,
     Unyank = 2,
 }
 
+impl FromSql<Integer, Pg> for VersionAction {
+    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
+        match <i32 as FromSql<Integer, Pg>>::from_sql(bytes)? {
+            0 => Ok(VersionAction::Publish),
+            1 => Ok(VersionAction::Yank),
+            2 => Ok(VersionAction::Unyank),
+            n => Err(format!("unknown version action: {}", n).into()),
+        }
+    }
+}
+
+impl ToSql<Integer, Pg> for VersionAction {
+    fn to_sql<W: Write>(&self, out: &mut Output<'_, W, Pg>) -> serialize::Result {
+        ToSql::<Integer, Pg>::to_sql(&(*self as i32), out)
+    }
+}
+
 #[derive(Debug, Clone, Copy, Queryable, Identifiable, Associations)]
 #[belongs_to(Version)]
-#[belongs_to(User, foreign_key = "owner_id")]
-#[belongs_to(ApiToken, foreign_key = "owner_token_id")]
+#[belongs_to(User, foreign_key = "user_id")]
+#[belongs_to(ApiToken, foreign_key = "api_token_id")]
 #[table_name = "version_owner_actions"]
 pub struct VersionOwnerAction {
     pub id: i32,
     pub version_id: i32,
-    pub owner_id: i32,
-    pub owner_token_id: i32,
+    pub user_id: i32,
+    pub api_token_id: Option<i32>,
     pub action: VersionAction,
     pub time: NaiveDateTime,
 }
+
+impl VersionOwnerAction {
+    pub fn all(conn: &PgConnection) -> QueryResult<Vec<VersionOwnerAction>> {
+        version_owner_actions::table.load(conn)
+    }
+
+    pub fn by_version_id_and_action(
+        conn: &PgConnection,
+        _version_id: i32,
```

Comment:
This is a really small consistency fix I forgot to mention last time I looked at this PR, but I think I'd like to see it fixed before merging. 

In Rust, the convention is to start parameter and variable names with underscores when they aren't used. So when I look at this signature, I think that `_version_id` and `_action` aren't going to be used in the body, but they are. I know we have to use some sort of name that doesn't conflict with the table's columns. In some other places, [like the `find_by_api_token` function](https://github.com/rust-lang/crates.io/pull/1700/files#diff-1cdb3b1876ff75ba470bd31a2dda81e0R50), we instead *end* the parameter name with an underscore.

Do you mind changing `_version_id` and `_action` in this function, and the 4 parameters in the `insert_version_owner_action` function, to end with underscores rather than starting with underscores?